kola/harness.go

```diff
@@ -382,7 +388,7 @@ func parseCLVersion(input string) (*semver.Version, error) {
 // runTest is a harness for running a single test.
 // outputDir is where various test logs and data will be written for
 // analysis after the test run. It should already exist.
-func runTest(h *harness.H, t *register.Test, pltfrm string) {
+func runTest(h *harness.H, t *register.Test, pltfrm string, flight platform.Flight) {
```

Comment:
What do you think about moving the `architecture` function in this file into `Flight`? The determination could be handled at it's instantiation which would also allow us to drop passing the `pltfrm` string here (and add an easy way to access the platform architecture from inside of tests in the future).

Reply:
`architecture` is also called from `filterTests`, which doesn't have a flight.  I agree that the arch stuff needs to be cleaned up, especially now that we can spell it `architecture` instead of `board`, but I think that's a larger project.